homeassistant/config.py

```diff
@@ -663,6 +666,27 @@ async def async_process_ha_core_config(hass: HomeAssistant, config: dict) -> Non
     if CONF_UNIT_SYSTEM in config:
         hac.units = get_unit_system(config[CONF_UNIT_SYSTEM])
 
+    async def load_translations(_event: Event) -> None:
+        language = hass.config.language
+        _LOGGER.debug(f"Loading translations for language: {language}")
+        await async_load_state_translations_to_cache(hass, language)
+
+    async def load_translations_for_component(event: Event) -> None:
+        component = event.data.get("component")
+        if component is None or "." in str(component):
+            _LOGGER.debug(
+                f"Skipping loading translations for an invalid component: {component}"
+            )
+            return
+        language = hass.config.language
+        _LOGGER.debug(
+            f"Loading translations for language: {hass.config.language} and component: {component}"
+        )
+        await async_load_state_translations_to_cache(hass, language, [component])
+
+    hass.bus.async_listen(EVENT_COMPONENT_LOADED, load_translations_for_component)
+    hass.bus.async_listen(EVENT_CORE_CONFIG_UPDATE, load_translations)
```

Comment:
I'm not convinced `config` should be responsible for maintaining the translation cache.
I think it could instead be handled `helpers/translations.py`, with a call from here to setup the listeners: https://github.com/home-assistant/core/blob/c433b251fa125789a2b8a68e67014d0eeb7bf6e0/homeassistant/bootstrap.py#L256-L259

Reply:
I have extracted `async_setup_load_listeners`, added it to `helpers/translation.py` and called from `bootstrap.py`